driver/config/config.go

```diff
@@ -1304,16 +1308,31 @@ func (p *Config) GetTSLCertificatesForAdmin(ctx context.Context) []tls.Certifica
 	)
 }
 
-func (p *Config) getTSLCertificates(daemon, certBase64, keyBase64, certPath, keyPath string) []tls.Certificate {
-	cert, err := tlsx.Certificate(certBase64, keyBase64, certPath, keyPath)
-
-	if err == nil {
+func (p *Config) getTLSCertificates(ctx context.Context, daemon, certBase64, keyBase64, certPath, keyPath string) CertFunc {
+	if certBase64 != "" && keyBase64 != "" {
+		cert, err := tlsx.CertificateFromBase64(certBase64, keyBase64)
+		if err != nil {
+			p.l.WithError(err).Fatalf("Unable to load HTTPS TLS Certificate")
+			return nil // reachable in unit tests when Fatalf is hooked
+		}
 		p.l.Infof("Setting up HTTPS for %s", daemon)
-		return cert
-	} else if !errors.Is(err, tlsx.ErrNoCertificatesConfigured) {
-		p.l.WithError(err).Fatalf("Unable to load HTTPS TLS Certificate")
+		return func(*tls.ClientHelloInfo) (*tls.Certificate, error) { return &cert, nil }
+	}
+	if certPath != "" && keyPath != "" {
+		errs := make(chan error, 1)
+		getCert, err := tlsx.GetCertificate(ctx, certPath, keyPath, errs)
+		if err != nil {
```

Comment:
Is `!errors.Is(err, tlsx.ErrNoCertificatesConfigured)` no longer a valid error state, or why was it ignored?

Reply:
`tlsx.ErrNoCertificatesConfigured` is only returned by `tlsx.Certificate()`, which has little bit of a clunky interface. Instead of having the whole logic governing where certificates are read from (disk or base64-encoded in config value) inside a call to a library function, I moved this decision into this function. We call either `tlsx.CertificateFromBase64` or `tlsx.GetCertificate` depending on which config keys are set. If none are set, we return nil (no TLS).

It might make sense to make this logic even more strict and explicitly fail when the user tries to configure both a path and a base64 value (or a mixture of the two).

Reply:
Thanks!